odoo/addons/base/ir/ir_ui_view.py

```diff
@@ -328,13 +328,22 @@ def _check_xml(self):
                     view_docs = view_docs[0]
                 validator = self._relaxng()
                 for view_arch in view_docs:
-                    version = view_arch.get('version', '7.0')
-                    if parse_version(version) < parse_version('7.0') and validator and not validator.validate(view_arch):
-                        for error in validator.error_log:
-                            _logger.error(tools.ustr(error))
-                        raise ValidationError(_('Invalid view definition'))
-                    if not valid_view(view_arch):
-                        raise ValidationError(_('Invalid view definition'))
+                    if view.type in ['form', 'kanban']:
+                        if not valid_view(view_arch):
+                            raise ValidationError(_('Invalid view definition'))
+                    if view.type in ['search', 'calendar', 'gantt', 'tree', 'graph', 'pivot']:
```

Comment:
> We can't added rng-based validation for 'form' and 'kanban' view because in that we are used html tages also.

I never told you to do that, I told you that view_validation should apply to every view type (which you have fixed and is great).

And I am telling you to move rng validation to view_validation as well, there is no reason for it to remain a special case in ir_ui_view when we have a more generic validation hook, and that maybe each view type can be validated with its own rng to avoid cross-contamination of the schema files and make them more self-contained and easier to read.